server/graphql/types/src/types/program/patient.rs

```diff
@@ -126,14 +136,59 @@ impl PatientNode {
                 None,
                 Some(filter),
                 self.allowed_ctx.clone(),
-            )?;
-        Ok(entries
+            )?
             .into_iter()
-            .map(|program_enrolment| ProgramEnrolmentNode {
+            .map(|program_row| ProgramEnrolmentNode {
                 store_id: self.store_id.clone(),
-                program_enrolment,
+                program_enrolment: program_row,
                 allowed_ctx: self.allowed_ctx.clone(),
             })
-            .collect())
+            .collect();
+        Ok(ProgramEnrolmentResponse::Response(
+            ProgramEnrolmentConnector {
+                total_count: usize_to_u32(nodes.len()),
+                nodes,
+            },
+        ))
+    }
+
+    pub async fn contact_traces(
```

Comment:
I'm not used to seeing the nested filter/pagination things within a graphql type like this.
Is this needed because you want to use the patient query to control access rights to this?
Managing nested FilterInputs etc, seems a bit complex, For example if you have page 1 for patients but want page 2 for their contact traces? I assume you'll be using it in a single patient context?
Interested to see what the query will like in the frontend...

Reply:
I actually use it for reports. For example: query all patients and then get the latest, single encounter for this patient...